todo/commands/complete.py

```diff
@@ -0,0 +1,160 @@
+        """Commands having to do with your completed tasks
+
+
+        \u200b
+        **Arguments**
+            - `indexes` Optional indexes to complete. If left at none the help command will be shown
+        """
+        indexes = [i - 1 for i in indexes]
+        data = await self.cache.get_user_data(ctx.author.id)
+        todos = data["todos"]
+        if not todos:
+            return await ctx.send(self._no_todo_message.format(prefix=ctx.clean_prefix))
+        completed = []
+        for index in indexes:
+            try:
+                completed.append((todos.pop(index))["task"])
+            except IndexError:
+                pass
+            except Exception as e:
+                self.log.error("Error in command 'todo complete'", exc_info=e)
+        amount = len(completed)
+        plural = "" if amount == 1 else "s"
+        await ctx.send(f"Completed {amount} todo{plural}")
+        data["completed"].extend(completed)
+        data["todos"] = todos
+        await self.cache.set_user_data(ctx.author, data)
+        await self._maybe_autosort(ctx.author)
+
+    @complete.command(name="delete", aliases=["del", "remove"])
+    async def complete_delete(self, ctx: commands.Context, *indexes: PositiveInt):
+        """Delete completed todos
+
+        This will remove them from your completed list
+
+        **Arguments**
+            - `indexes` A list of integers for the indexes of your completed todos
+        """
+        indexes = [i - 1 for i in indexes]
+        completed = await self.cache.get_user_item(ctx.author, "completed")
+        if not completed:
+            return await ctx.send(
+                self._no_completed_message.format(prefix=ctx.clean_prefix)
+            )
+        for index in indexes:
+            try:
+                completed.pop(index)
+            except IndexError:
+                pass
+            except Exception as e:
+                self.log.error("Exception in command 'todo complete delete'", exc_info=e)
+        amount = len(indexes)
+        plural = "" if amount == 1 else "s"
+        await ctx.send(f"Deleted {amount} completed todo{plural}")
+        await self.cache.set_user_item(ctx.author, "completed", completed)
+
+    @complete.command(name="deleteall", aliases=["delall", "removeall"])
+    async def complete_remove_all(self, ctx: commands.Context, confirm: bool = False):
+        """Remove all of your completed todos
+
+
+        \u200b
+        **Arguments**
+            - `confirm` Skips the confirmation check. Defaults to False
+        """
+        if not confirm:
+            msg = await ctx.send(
+                "Are you sure you would like to remove all of your completed todos? (y/N)"
+            )
+            pred = MessagePredicate.yes_or_no(ctx)
+            try:
+                umsg = await self.bot.wait_for("message", check=pred)
+            except asyncio.TimeoutError:
+                pass
```

Comment:
I do this just because it gets caught by `if not pred.result` as `MessagePredicate.result` starts as `None` and will therefore be caught by that if statement. Feels nicer for the user imo